### PR TITLE
Convert AUTHOR and CONTRIBUTE to rst files and add them to the doc

### PR DESCRIPTION
**Pull request by xuhdev**
---

## Checklist:

<details open>

<summary></summary>

- Does this pull request close an issue? We encourage you to open an issue first if this pull request (PR) is not a
  minor change.
  - [ ] No
    - [ ] The change in this pull request is minor.
  - [x] Yes
    - Partly on #126

For the following questions, only check the boxes that are applicable.

- [x] Ensure that the pull request text above the horizontal line is descriptive.
- [x] Add/update relevant tests.
- [x] Add/update relevant documents.

- Do you have triage permission of this repository?
  - [ ] No
    - You are good.
  - [x] Yes
    - Does this pull request close an issue?
      - [ ] No, this pull request also acts as an issue by itself.
        - [ ] Assign yourself.
        - [ ] Label this pull request.
        - [ ] Put a milestone.
        - [ ] Put this pull request under "In Progress" or "Under Review" pipeline on ZenHub.
        - [ ] Put this pull request under the appropriate epic on ZenHub.
        - [ ] Put an estimate on ZenHub.
        - [ ] Add dependency relationship with other issues/pull requests on ZenHub.
      - [x] Yes, this pull request addresses an issue and does not act as an issue.
        - [x] Connect this pull request to the underlying issue on ZenHub.
        - [x] DON'T:
          - DON'T assign yourself or anyone else.
          - DON'T label.
          - DON'T put a milestone.
          - DON'T put this pull request under any epic on ZenHub.
          - DON'T put an estimate on ZenHub.
    - [x] When the pull request is ready, request review.
</details>

### CODE REVIEW

File: AUTHORS.rst
Comment (xuhdev) [inline AUTHORS.rst:10]: @djalova Does this sound good to you?

File: CONTRIBUTING.rst
Comment (edwardleardi) [inline CONTRIBUTING.rst]: Missed the subheadings here:
`## Where to Import a Symbol?`, `## Continuous Integration (CI)`, `## Docs`

**Review by edwardleardi — CHANGES_REQUESTED** [review]
We still want to convert the README as well right? We'll do that in a separate PR then?

**Comment by xuhdev** [comment]
> We still want to convert the README as well right? We'll do that in a separate PR then?

Yeah, we'll do it separately. Just to prevent the PR from being too big and too hard to review

File: CONTRIBUTING.rst
Comment (edwardleardi) [inline CONTRIBUTING.rst]:
```suggestion
We prefer keeping CI configuration files, namely :file:`.github/workflows/*` and :file:`.gitlab-ci.yml`, simple and unscrambled.
```

**Review by edwardleardi — CHANGES_REQUESTED** [review]
Sorry one other small thing I noticed

**Review by edwardleardi — APPROVED** [review]
LGTM, feel free to merge if you're waiting for Daniel to reply to your comment

File: AUTHORS.rst
Comment (djalova) [reply AUTHORS.rst:10]: Sounds good

**Pull request merged**
